interp/src/interpreter/interpret_control.rs

```diff
@@ -44,12 +44,42 @@ fn eval_seq(
 /// Interpret Par
 
 fn eval_par(
-    _p: &ir::Par,
-    _continuous_assignments: &[ir::Assignment],
-    mut _env: InterpreterState,
-    _comp: &ir::Component,
+    p: &ir::Par,
+    continuous_assignments: &[ir::Assignment],
+    mut env: InterpreterState,
+    comp: &ir::Component,
 ) -> FutilResult<InterpreterState> {
-    todo!("par control operator")
+    let mut sts = Vec::new();
+
+    for st in &p.stmts {
+        sts.push(interpret_control(
+            st,
+            continuous_assignments,
+            //change fork to do fork first and then tail
+            env.fork(),
+            comp,
+        )?);
+    }
+
+    sts.push(env);
+
+    let mut tl = 0;
+    let mut sms = Vec::new();
+
+    let mut final_st = sts.remove(0);
+
+    for is in sts {
+        if is.clk > tl {
+            tl = is.clk;
+        }
+
+        sms.push(is.pv_map);
+    }
+
+    final_st.pv_map = final_st.pv_map.merge_many(sms);
```

Comment:
seems like you could just not push `env` into `sms` and use it here rather than incurring the cost of removing the zeroth index and shifting the values in the vec. Any reason why this was not the approach taken?

Reply:
fixed

Reply:
I don't see the change. Have you committed it yet?